### ISSUE TITLE
Metal golden-logit snapshot test failing on main: logits 2x expected

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#31]
## Summary

`test_metal_qwen35_golden_logit_snapshot_forward_step_token_42_pos_0` is
failing on `main` (and has been since at least the QuaRot step-4 work).
The actual logits are roughly **2×** the expected golden values, suggesting
a forward-path change went in without re-snapshotting.

## Reproducer

```bash
cd crates/inference
cargo test -p lattice-inference --lib --features f16,metal-gpu \
  test_metal_qwen35_golden_logit_snapshot_forward_step_token_42_pos_0
```

## Observed vs expected

```text
actual   = [-45.240494, 0.0, 45.240494, -45.240494, 0.0,
            45.240494, -45.240494, 0.0, 45.240494, -45.240494]
expected = [-22.621626, 0.0, 22.621626, -22.621626, 0.0,
            22.621626, -22.621626, 0.0, 22.621626, -22.621626]
max_abs_diff = 22.618868
```

`45.240494 ≈ 2 × 22.621626` on every position, including the zeros — a clean
multiplicative drift, not noise. Smells like a missing-or-doubled scale
factor somewhere on the final logit path (final RMSNorm gain, tied-embedding
matmul, or LM head projection). Could also be a doubled `(1 + γ)` fusion
applied to a layer that already has its norm in the fixture set to `1.0`.

## Provenance

- Test added in `d4b9839` ("feat: initial release — pure Rust inference engine")
  with the `22.6216266` golden values.
- Test source: `crates/inference/src/forward/metal_qwen35.rs:10718` (current
  line; the function name and assertion are unchanged since the initial commit).
- Fixture: `tiny_metal_qwen35_fixture` (1-layer FullAttention, hidden=512,
  vocab=64, tied embeddings, `final_norm = [1.0; hidden]`).
- Confirmed reproducible on `main` (commit `09cd554` and earlier) before
  any step-4 work, so it is independent of ADR-044 step 4 PRs.

## Suggested triage

1. `git bisect` between `d4b9839` and current `main` on the failing test
   with `--features f16,metal-gpu` — the bisect should land on a single
   forward-path commit.
2. Diagnose whether the doubling is in (a) embedding lookup, (b) the
   single full-attention block, (c) the residual add path, (d) the final
   RMSNorm + matmul against the tied embedding. The `2×` pattern + the
   preserved zeros narrows it to a linear scale somewhere on the post-
   residual / pre-logit path.
3. Once the cause is identified, decide whether the new behavior is
   intentional (re-snapshot) or a regression (fix).

## Why this is blocking

Not blocking — the failure is well-bounded to one test, and the rest of
the Metal suite (582 other tests) is green. Step 4b (`PR feat/quarot-step-4b-metal-q4-perplexity`)
explicitly skips this test in its commit message because the failure
pre-dates that branch.

Filing this so it doesn't keep getting noted-and-skipped across PRs.

> _Issue body authored by Claude (Anthropic agent) on behalf of @ohdearquant._
